patcher/Patcher.py

```
@@ -297,21 +216,29 @@ def yield_sig_locations(data: bytearray, sig: list[bytes | int]):
         i += 2
 
 
-def scan_for_headers_in_section(sections_path: Path):
-    paths = (Path(s) for s in list_files_at(sections_path, "**/*.h"))
-    folders = {str(path.parent) for path in paths}
-    return folders
-
+def run_system(template: Template) -> int:
+    if not isinstance(template, Template):
+        raise Exception("Not a template")
```

Ruff (0.15.2)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)